File: src/main/java/com/github/thorben/webnd/application/controller/HomeController.java
package com.github.thorben.webnd.application.controller;

import com.github.thorben.webnd.application.tech.MessageBroker;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.annotation.Resource;

@Controller
@RequestMapping("/")
public class HomeController {

	@Resource(name = "messageBroker")
	private MessageBroker messageBroker;

	@GetMapping
	public String index(Model model) {
		messageBroker.applyMessages(model);
		return "index";
	}

	@GetMapping("/characters")
	public String characterHome() {
		return "characters";
	}

}

File: src/main/kotlin/com/github/thorben/webnd/application/controller/HomeController.kt
package com.github.thorben.webnd.application.controller

import com.github.thorben.webnd.application.tech.StatelessMessageBroker
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import javax.servlet.http.HttpServletRequest

@Controller
@RequestMapping("/")
class HomeController {

	private val messageBroker = StatelessMessageBroker()

	@GetMapping
	fun index(httpServletRequest: HttpServletRequest, model: Model): String {
		messageBroker.applyMessages(httpServletRequest, model)
		return "index"
	}

	@GetMapping("/characters")
	fun characterHome(): String {
		return "characters"
	}
}